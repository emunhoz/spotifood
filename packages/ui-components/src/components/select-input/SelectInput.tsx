import React, { ChangeEvent } from 'react'
import { Label } from '../../'
import * as S from './SelectInput.style'

interface ISelectInput {
  disabled?: boolean
  /**
   * Input value
   */
  value?: string
  /**
   * Placeholder for input
   */
  placeholder?: string
  /**
   * Label for input
   */
  label?: string
  /**
   * onChange callback
   */
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void
  /**
   * Children options tag
   */
  children?: React.ReactNode
}

export const SelectInput = ({
  value,
  onChange,
  placeholder,
  disabled,
  label = '',
  children,
  ...rest
}: ISelectInput) => (
  <S.SelectInputWrapper>
    <S.Label>
      <Label htmlFor={label} value={label} />
      <S.SelectInput name={label} value={value} onChange={onChange} {...rest}>
        {children}
      </S.SelectInput>
    </S.Label>
  </S.SelectInputWrapper>
)
