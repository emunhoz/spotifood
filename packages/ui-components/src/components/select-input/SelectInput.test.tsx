import React from 'react'
import { render } from '../../utils/test-utils'
import { SelectInput } from './SelectInput'

describe('[Component]: SelectInput', () => {
  test('render the component', () => {
    const { asFragment } = render(
    <SelectInput onChange={console.log} label="Select example" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
