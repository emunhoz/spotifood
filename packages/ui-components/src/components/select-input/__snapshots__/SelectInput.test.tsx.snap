// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: SelectInput render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-ezrdKe"
  >
    <div
      class="sc-kLgntA cOPeoQ"
    >
      <label
        class="sc-hBEYos iAuSGj"
        for="Select example"
      >
        Select example
      </label>
      <select
        class="sc-bYEvPH kzdBAi"
        name="Select example"
      />
    </div>
  </div>
</DocumentFragment>
`;
